paddle/phi/api/generator/dist_api_gen.py

```diff
@@ -81,6 +81,31 @@
   }}
 """
 
+NCCL_COMMCONTEXT_INIT = """
+#if defined(PADDLE_WITH_NCCL) || defined(PADDLE_WITH_RCCL)
+  const auto & comm_context_manager = phi::distributed::CommContextManager::GetInstance();
+  phi::distributed::NCCLCommContext* comm_context = nullptr;
+  if (comm_context_manager.Has(std::to_string(ring_id))) {{
+    comm_context = static_cast<phi::distributed::NCCLCommContext *>(
+          comm_context_manager.Get(std::to_string(ring_id)));
+    PADDLE_ENFORCE_NE(
+        comm_context,
+        nullptr,
+        common::errors::Unavailable(
+            "NCCLCommContext is nullptr, collective op should "
+            "has ring_id attr."));
+    auto kernel_res = phi::KernelFactory::Instance().SelectKernelOrThrowError(
+        "{}", {{kernel_backend, kernel_layout, kernel_data_type}}, true);
+    if (FLAGS_low_precision_op_list) {{
+      phi::KernelFactory::Instance().AddToLowPrecisionKernelList("{}", kernel_data_type);
+    }}
+    Backend act_kernel_backend = kernel_res.has_fallback_cpu ? Backend::CPU : kernel_backend;
+    auto* dev_context = GetDeviceContextByBackend(act_kernel_backend);
+    dev_context->SetCommContext(comm_context);
```

Comment:
TODO: get `dev_context` from comm_context

Reply:
get it